src/main/java/com/faforever/client/chat/ChannelTabController.java

```diff
@@ -235,6 +249,35 @@ public void setChannel(Channel channel) {
     searchFieldContainer.visibleProperty().bind(searchField.visibleProperty());
     closeSearchFieldButton.visibleProperty().bind(searchField.visibleProperty());
     addSearchFieldListener();
+    topicPane.managedProperty().bind(topicPane.visibleProperty());
+    updateChannelTopic();
+    JavaFxUtil.addListener(channel.topicProperty(), observable -> Platform.runLater(this::updateChannelTopic));
+  }
+
+  private void updateChannelTopic() {
+    boolean hasTopic = channel.getTopic() != null;
+    topicPane.setVisible(hasTopic);
+    if (!hasTopic) {
+      return;
+    }
+    String topic = channel.getTopic() == null ? "" : channel.getTopic();
+    Arrays.stream(topic.split("\\s"))
+        .forEach(word -> {
+          if (URL_REGEX_PATTERN.matcher(word).matches()) {
```

Comment:
Is there any code that finds URLs already? Does chat area do that?

Reply:
yes but it is done in Java Script

Reply:
moved it to PlatformService